Design.md

```diff
@@ -0,0 +1,187 @@
+---
+authors: Brian Green (bgreen007@cox.net)
+state: draft
+---
+
+## What
+Job worker service that provides an API to run arbitrary Linux processes.
+
+## Details
+
+
+### Security
+Use mTLS for gRPC communication.
+
+Create a self-signed root certificate authority (CA) to sign the certificates used by the clients and server. This CA will also be used to verify the authenticity of the certs during gRPC authentication. Using RSA 2048.
+
+Required files: Server
+| File | Description |
+| --- | --- |
+| server_cert.pem | the server's certificate (public key) |
+| server_key.pem | the server's private key |
+| root_ca.pem | the certificate of the CA that can verify the client's certificate |
+
+Required files: Client:
+| File | Description |
+| --- | --- |
+| client_cert.pem | the client's certificate (public key). |
+| client_key.pem | the client's private key. |
+| root_ca.pem | the certificate of the CA that can verify the server's certificate. |
+
+Use CloudFlare's `cfssl` CLI to create and sign all of the certs mentioned above.
+
+#### Creating the self-signed root CA
+`cfssl selfsign -config cfssl.json --profile rootca "Greenbeans Dev CA" csr.json | cfssljson -bare root`
+
+[cfssl.json](cfssl.json)
+[csr.json](csr.json)
+
+# Creating and signing the server certificate:
+```
+cfssl genkey csr.json | cfssljson -bare server
+cfssl sign -ca root.pem -ca-key root-key.pem -config cfssl.json -profile server server.csr | cfssljson -bare server
+
+```
+
+# Creating and signing the client(s) certificates:
+```
+cfssl genkey csr.json | cfssljson -bare client
+cfssl sign -ca root.pem -ca-key root-key.pem -config cfssl.json -profile client client.csr | cfssljson -bare client
+```
+Multiple clients are supported, but each client will require it own certificate. The calls above can easily be modified to include additional client information (name):
+```
+cfssl genkey csr.json | cfssljson -bare client_alex
+cfssl sign -ca root.pem -ca-key root-key.pem -config cfssl.json -profile client client_alex.csr | cfssljson -bare client_alex
+```
+The CLI will include an optional parameter to specify the `client` key name, otherwise will default to 'client'.
```

Comment:
I'm not sure I quite grasp what this line is referring to.

Reply:
an optional command line parameter that allows you to specify a different key and cert for the client other than the default value it tries to load (client_cert.pem and client_key.pem)